pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/model/activity/definition/FormActivityDef.java

```diff
@@ -188,6 +188,13 @@ public QuestionDef getQuestionByStableId(String stableId) {
         return stableIdToQuestion.get(stableId);
     }
 
+    public Set<QuestionDef> getAllQuestions() {
+        if (stableIdToQuestion == null) {
+            stableIdToQuestion = buildStableIdToQuestionMap();
+        }
+        return new HashSet<>(stableIdToQuestion.values());
+    }
+
     private Map<String, QuestionDef> buildStableIdToQuestionMap() {
         return getAllSections().stream()
                 .flatMap(section -> section.getBlocks().stream())
```

Comment:
Is it possible if the blocks section is null?
If yes, then I'd suggest using
```
getAllSections().stream()
                .map(Section::getBlocks)
                .flatMap(Collection::stream)
```
to avoid possible NPE

Reply:
blocks field is annotated at notnull. We should be OK.